packages/intrepid2/src/Orientation/Intrepid2_OrientationToolsDefCoeffMatrix_HGRAD.hpp

```diff
@@ -201,64 +194,59 @@ getCoeffMatrix_HGRAD(OutputViewType &output,
   //
 
   // construct Psi and Phi  matrices.  LAPACK wants left layout
-  Kokkos::View<ScalarType**,Kokkos::LayoutLeft,ExecutionSpace>
-  PsiMat("PsiMat", ndofSubcell, ndofSubcell),
-  PhiMat("PhiMat", ndofSubcell, ndofSubcell);
-
-  auto cellTagToOrdinal = Kokkos::create_mirror_view_and_copy(typename ExecutionSpace::memory_space(), cellBasis.getAllDofOrdinal());
-  auto subcellTagToOrdinal = Kokkos::create_mirror_view_and_copy(typename ExecutionSpace::memory_space(), subcellBasis.getAllDofOrdinal());
-
-  Kokkos::parallel_for( Kokkos::RangePolicy<ExecutionSpace>(0, ndofSubcell),
-        KOKKOS_LAMBDA (const int i){
+  Kokkos::DynRankView<value_type,Kokkos::LayoutLeft,host_device_type>
+    PsiMat("PsiMat", ndofSubcell, ndofSubcell),
+    PhiMat("PhiMat", ndofSubcell, ndofSubcell);
+  
+  auto cellTagToOrdinal = cellBasis.getAllDofOrdinal();
+  auto subcellTagToOrdinal = subcellBasis.getAllDofOrdinal();
+
+  for (ordinal_type i=0;i<ndofSubcell;++i) {
```

Comment:
Here again with the for v. parallel_for.

Reply:
See the above comment.